src/poetry/repositories/repository_pool.py

```diff
@@ -71,13 +76,23 @@ def has_repository(self, name: str) -> bool:
         return name.lower() in self._repositories
 
     def repository(self, name: str) -> Repository:
+        return self._get_prioritized_repository(name).repository
+
+    def get_priority(self, name: str) -> Priority:
+        return self._get_prioritized_repository(name).priority
+
+    def _get_prioritized_repository(self, name: str) -> PrioritizedRepository:
         name = name.lower()
         if self.has_repository(name):
-            return self._repositories[name].repository
+            return self._repositories[name]
         raise IndexError(f'Repository "{name}" does not exist.')
 
     def add_repository(
-        self, repository: Repository, default: bool = False, secondary: bool = False
+        self,
+        repository: Repository,
+        default: bool = False,
+        secondary: bool = False,
+        priority: Priority = Priority.PRIMARY,
```

Comment:
```suggestion
        secondary: bool = False,
        *,
        priority: Priority = Priority.PRIMARY,
```

(makes removing `default` and `secondary` easier in the future)